crates/sargon/src/wrapped_radix_engine_toolkit/low_level/v2/intent_header_v2.rs

```diff
@@ -0,0 +1,175 @@
+use crate::prelude::*;
+
+/// Represents the header of an intent in V2, containing network ID,
+/// epoch range, optional proposer timestamps, and an intent discriminator.
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[display("{} intent_discriminator: {}", network_id, intent_discriminator)]
+pub struct IntentHeaderV2 {
+    pub network_id: NetworkID,
+    pub start_epoch_inclusive: Epoch,
+    pub end_epoch_exclusive: Epoch,
+    pub min_proposer_timestamp_inclusive: Option<Instant>,
+    pub max_proposer_timestamp_exclusive: Option<Instant>,
+    pub intent_discriminator: IntentDiscriminator,
+}
+
+impl IntentHeaderV2 {
+    /// Creates a new `IntentHeaderV2`
+    ///
+    /// # Panics
+    /// Panics if `end_epoch_exclusive < start_epoch_inclusive`
+    pub fn new(
+        network_id: NetworkID,
+        start_epoch_inclusive: impl Into<Epoch>,
+        end_epoch_exclusive: impl Into<Epoch>,
+        min_proposer_timestamp_inclusive: Option<Instant>,
+        max_proposer_timestamp_exclusive: Option<Instant>,
+        intent_discriminator: impl Into<IntentDiscriminator>,
+    ) -> Self {
+        let start_epoch_inclusive = start_epoch_inclusive.into();
+        let end_epoch_exclusive = end_epoch_exclusive.into();
+        let intent_discriminator = intent_discriminator.into();
+        assert!(
+            end_epoch_exclusive >= start_epoch_inclusive,
+            "End epoch MUST be greater than or equal start epoch."
+        );
+
+        if let (Some(min_ts), Some(max_ts)) = (
+            min_proposer_timestamp_inclusive,
+            max_proposer_timestamp_exclusive,
+        ) {
+            assert!(
+                max_ts >= min_ts,
+                "Max proposer timestamp MUST be greater than or equal min proposer timestamp."
+            );
+        }
```

Comment:
Should we add additional checks for the proposer timestamps? 🤔

Reply:
proposer timestamps should be in the epoch window I think.